### PR TITLE
[ipa-4-12] ipatests: configure gating and nightly tests on ipa-4-12 branch

### PR DESCRIPTION
**Pull request by flo-renaud**
Update the pipelines for ipa-4-12 branch:
- run tests on fedora 40
- use the vagrant image freeipa/ci-ipa-4-12-f40

### CODE REVIEW

File: ipatests/prci_definitions/gating.yaml [hunk 1]
```diff
@@ -21,316 +21,316 @@ topologies:
...
     job:
       class: Build
       args:
         git_repo: '{git_repo}'
         git_refspec: '{git_refspec}'
-        template: &ci-master-latest
-          name: freeipa/ci-master-f39
+        template: &ci-ipa-4-12-latest
+          name: freeipa/ci-ipa-4-12-f40
           version: 0.0.1
```
Comment (f-trivino) [inline ipatests/prci_definitions/gating.yaml:34]: shouldn't be:

`version: 0.0.0 ?`

image

File: ipatests/prci_definitions/temp_commit.yaml [hunk 2]
```diff
@@ -47,28 +47,28 @@ topologies:
...
     job:
       class: Build
       args:
         git_repo: '{git_repo}'
         git_refspec: '{git_refspec}'
-        template: &ci-master-latest
-          name: freeipa/ci-master-f39
+        template: &ci-ipa-4-12-latest
+          name: freeipa/ci-ipa-4-12-f40
           version: 0.0.1
```
Comment (f-trivino) [inline ipatests/prci_definitions/temp_commit.yaml:60]: version 0.0.1 ?

File: ipatests/prci_definitions/nightly_ipa-4-12_latest.yaml [hunk 3]
```diff
@@ -41,2039 +41,2039 @@ topologies:
...
     job:
       class: Build
       args:
         git_repo: '{git_repo}'
         git_refspec: '{git_refspec}'
-        template: &ci-master-latest
-          name: freeipa/ci-master-f39
+        template: &ci-ipa-4-12-latest
+          name: freeipa/ci-ipa-4-12-f40
           version: 0.0.1
```
Comment (f-trivino) [inline ipatests/prci_definitions/nightly_ipa-4-12_latest.yaml:54]: version: 0.0.0 ?

**Review by f-trivino — CHANGES_REQUESTED** [review]
@flo-renaud not sure if the freeipa/ci-ipa-4-12-f40 version should be 0.0.0

File: ipatests/prci_definitions/gating.yaml (hunk 1 above)
Comment (flo-renaud) [reply ipatests/prci_definitions/gating.yaml:34]: @f-trivino 
the latest version is 0.0.1:
image

Comment (f-trivino) [reply ipatests/prci_definitions/gating.yaml:34]: I see it now, thanks

**Review by f-trivino — APPROVED** [review]
LGTM, please rebase it.

**Comment by flo-renaud** [comment]
ipa-4-12:

* 58154be74fa950b3356712e60687930abb6480f1 ipatests: configure gating and nightly tests on ipa-4-12 branch
* 4521fe5f9125c74b4ad6e4e51f8c66c009079281 ipatests: mark test_ca_show_error_handling as xfail
* 60c127d197f79fa4ed612f7173e752d156885415 ipatests: fix / permissions to allow ssh with private key

**Pull request closed without merging**